### PR TITLE
Rewrite tutorial

### LINKED ISSUE #3: Errors
This code has lots of errors and missing parts and isn't working at all. I will be rewriting this example so that it actually works within the next couple of days

### PR DESCRIPTION
**Pull request by thewilloftheshadow**
Fixes #3 

A much simpler implementation of the mebots API in GroupMe

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,4 +23,4 @@ heroku config:set BOT_TOKEN=0123456789abcdef
 ```
 
 ## Author
-[Erik Boesen](https://github.com/ErikBoesen)
+[Will Shadow](https://github.com/thewilloftheshadow)
```
Comment (ErikBoesen) [inline README.md:27]: I did write the original code so I would prefer that we both be listed here.

File: index.js [hunk 2]
```diff
@@ -1,33 +1,44 @@
...
 });
 
-server = http.createServer(function (req, res) {
-    req.chunks = [];
-    req.on('data', function (chunk) {
-        req.chunks.push(chunk.toString());
-    });
+// Use this route as the callback URL for the bot
+// Example: https://shadow.herokuapp.com/gm
+app.post('/gm', async (req, res)  => {
```
Comment (ErikBoesen) [inline index.js]: Hmm, I don't really like this route naming. I usually use the `/` route, which is not ideal, but perhaps we should choose something more clear, for example `/receive`? `gm` is kind of ambiguous.

File: index.js [hunk 3]
```diff
@@ -1,33 +1,44 @@
...
+  console.log(req.body)
+  let cmd = req.body.text
+  if(cmd === "/ping") sendmsg("Pong!", req.body.group_id)
```
Comment (ErikBoesen) [inline index.js]: Spacing

File: package.json [hunk 4]
```diff
@@ -1,28 +1,25 @@
 {
-  "name": "bot-tutorial",
-  "version": "0.1.2",
-  "description": "A simple GroupMe bot written in Node.js",
+  "name": "mebots-nodejs-example",
```
Comment (ErikBoesen) [inline package.json]: Word, good catch

**Review by ErikBoesen — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (thewilloftheshadow) [reply index.js]: ahh that's true. it stood for groupme in the bot I took this from, but a better name is good

File: index.js (hunk 3 above)
Comment (thewilloftheshadow) [reply index.js]: my biggest struggle

**Review by thewilloftheshadow — COMMENTED** [review]
Made some changes, adding spacing and renaming routes, and some additions to package.json

File: README.md (hunk 1 above)
Comment (thewilloftheshadow) [reply README.md:27]: :+1: true, that makes sense

File: index.js [hunk 5]
```diff
@@ -1,33 +1,48 @@
...
-    req.on('data', function (chunk) {
-        req.chunks.push(chunk.toString());
-    });
+// Use this route as the callback URL for the bot
+// Example: https://shadow.herokuapp.com/receive
+app.post('/receive', async (req, res)  => {
+  console.log(req.body)
+  let cmd = req.body.text
+  
+  if(cmd === "/ping"){
```
Comment (ErikBoesen) [inline index.js]: Again, spacing. `if (cmd === "/ping") {`. Also be consistent with quote type

File: index.js [hunk 6]
```diff
@@ -1,33 +1,48 @@
...
+const express = require("express");
+const app = express();
 
-router = new director.http.Router({
-    '/' : {
-        post: bot.respond,
-        get: ping
-    }
+// Library to send POST requests simply
+const axios = require('axios')
```
Comment (ErikBoesen) [inline index.js:10]: I am hesitant to complicate this example with an arbitrary library like this; I feel that we should stick to the most basic implementation wherever possible so consider maybe not using axios

**Review by ErikBoesen — CHANGES_REQUESTED** [review]

**Comment by ErikBoesen** [comment]
Also note that you can say "Fixes #3" in the description of your PR and it will close the issue when I merge your PR.

File: index.js (hunk 6 above)
Comment (thewilloftheshadow) [reply index.js:10]: this is the most basic implementation. without using axios or a similar library, we get code like this:
```
const https = require('https');

function httpsPost({body, ...options}) {
    return new Promise((resolve,reject) => {
        const req = https.request({
            method: 'POST',
            ...options,
        }, res => {
            const chunks = [];
            res.on('data', data => chunks.push(data))
            res.on('end', () => {
                let body = Buffer.concat(chunks);
                switch(res.headers['content-type']) {
                    case 'application/json':
                        body = JSON.parse(body);
                        break;
                }
                resolve(body)
            })
        })
        req.on('error',reject);
        if(body) {
            req.write(body);
        }
        req.end();
    })
}
```

**Pull request merged**